### PR TITLE
refactor: (TransferList) CC => FC

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | refactor[Transfer]: CC => FC |
| 🇨🇳 Chinese | refactor[Transfer]: CC => FC |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/transfer/ListBody.tsx [hunk 1]
```diff
@@ -58,10 +58,7 @@ class ListBody<RecordType extends KeyWiseTransferItem> extends React.Component<
     if (mergedPagination) {
       // Calculate the page number
       const maxPageCount = Math.ceil(filteredRenderItems.length / mergedPagination.pageSize);
-
-      if (current > maxPageCount) {
-        return { current: maxPageCount };
-      }
+      return { current: Math.min(current, maxPageCount) };
```
Comment (zombieJ) [inline components/transfer/ListBody.tsx:61]: 重构要避免不等价的变更

File: components/transfer/__tests__/list.test.tsx [hunk 2]
```diff
@@ -35,25 +33,6 @@ describe('Transfer.List', () => {
     ).toBeTruthy();
   });
 
-  it('when component has been unmounted, componentWillUnmount should be called', () => {
```
Comment (zombieJ) [inline components/transfer/__tests__/list.test.tsx:38]: 这边要测的是 unmount 清理了 timeout 的 effect，不能直接删掉~

File: components/transfer/ListBody.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/transfer/ListBody.tsx:61]: 这里我 revert 一下

File: components/transfer/__tests__/list.test.tsx (hunk 2 above)
Comment (li-jia-nan) [reply components/transfer/__tests__/list.test.tsx:38]: timeout 没用到，重构成 FC 之后已经没有 timeout 了

Comment (zombieJ) [reply components/transfer/__tests__/list.test.tsx:38]: 嗯，看了一下。源码里这个 timeout 也没啥用

**Review by zombieJ — APPROVED** [review]

**Comment by li-jia-nan** [comment]
- #39893

**Pull request merged**